pkg/synchromanager/clustersynchro_manager.go

```diff
@@ -286,6 +312,22 @@ func (manager *Manager) reconcileCluster(cluster *clusterv1alpha2.PediaCluster)
 		}
 	}
 
+	if cluster.Spec.ShardingName != manager.shardingName {
+		if *cluster.Status.ShardingName == manager.shardingName {
```

Comment:
By the previous rule, `status.ShardingName` must be equal to `manager.shardingName` when `cluster.Spec.ShardingName != manager.shardingName`